model/textparse/openmetricsparse.go

```diff
@@ -294,17 +320,52 @@ func (p *OpenMetricsParser) CreatedTimestamp() *int64 {
 		// We got a CT line here, but let's search if CT line is actually for our series, edge case.
 		peekWithoutNameLsetHash, _ = peekedLset.HashWithoutLabels(buf, labels.MetricName, "le", "quantile")
 		if peekWithoutNameLsetHash != currFamilyLsetHash {
-			// CT line for a different series, for our series no CT.
+			// Found CT line for a different series, for our series no CT.
+			p.resetCTParseValues(resetLexer)
 			return nil
 		}
 
 		// All timestamps in OpenMetrics are Unix Epoch in seconds. Convert to milliseconds.
 		// https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#timestamps
-		ct := int64(peek.val * 1000.0)
+		ct := int64(p.val * 1000.0)
+		p.setCTParseValues(ct, currFamilyLsetHash, currName, true, resetLexer)
 		return &ct
 	}
 }
 
+// setCTParseValues sets the parser to the state after CreatedTimestamp method was called and CT was found.
+// This is useful to prevent re-parsing the same series again and early return the CT value.
+func (p *OpenMetricsParser) setCTParseValues(ct int64, ctHashSet uint64, visitedName string, skipCTSeries bool, resetLexer *openMetricsLexer) {
+	p.ct = ct
+	p.l = resetLexer
+	p.ctHashSet = ctHashSet
+	p.visitedName = visitedName
+	p.skipCTSeries = skipCTSeries
+}
+
+// resetCtParseValues resets the parser to the state before CreatedTimestamp method was called.
+func (p *OpenMetricsParser) resetCTParseValues(resetLexer *openMetricsLexer) {
+	p.l = resetLexer
+	p.ct = 0
+	p.ctHashSet = 0
+	p.visitedName = ""
+	p.skipCTSeries = true
+}
+
+// findBaseMetricName returns the metric name without reserved suffixes such as "_created",
+// "_sum", etc. based on the OpenMetrics specification found at
+// https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md.
+// If no suffix is found, the original name is returned.
+func findBaseMetricName(name string) string {
+	suffixes := []string{"_created", "_count", "_sum", "_bucket", "_total", "_gcount", "_gsum", "_info"}
+	for _, suffix := range suffixes {
+		if strings.HasSuffix(name, suffix) {
+			return strings.TrimSuffix(name, suffix)
+		}
+	}
+	return name
+}
```

Comment:
Not sure if we can do that without checking the appropriate types.

E.g. `_total` is only a suffix for counters, but it is still possible to have other metrics with this suffix. As far as I understood there is nothing in OM that prevents that from happening 🤔

Reply:
Yes, that's true
This function is used in one place atm (beginning of created timestamp function). otherwise we check with hashsets like before.

There's a weird case where if we compare a metric line that had no extra labels with a created line, it would generate the same hash (because we don't include __name__ when comparing hashsets)

Reply:
I feel there is a way to remove this logic.

We use `currName` to figure family name essentially, but isn't this parsed somewhere from metadata? 🤔

Reply:
I'd like to find a way too to get rid of this function. 

But even if we find the currName won't it have suffixes attached to it?

Reply:
Most of the time yes, but unfortunately all metadata fields are optional in OM, including type. But bartek has a good point, we don't need to do any of this if metadata was set, and just call this function if metadata isn't present.


This brings another problem though, if we call this function it means we don't know the type and therefore we don't know which suffixes we were supposed to remove 😬

Reply:
> but it is still possible to have other metrics with this suffix and it will be part of the metric name. 

but might still leave room for this

Reply:
We will never be perfect, but it would be nice to have exact test case for where we "know" we will be wrong. Ideally "wrong" means failure, not corrupted CT (:

Reply:
okay to better illustrate why we have this function I've updated tests
if you replace line 280 with `if currName == p.visitedName && currFamilyLsetHash == p.ctHashSet && p.visitedName != "" && p.ctHashSet > 0 && p.ct > 0 {` 

and debug it with `TestOpenMetricsParse` 

and focus on `currFamilyLsetHash` vs `p.ctHashSet` you'll notice we'll have the same hash in this case and it would incorrectly early return CT